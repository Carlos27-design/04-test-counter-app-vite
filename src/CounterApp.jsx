import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleSum = () => {
    // console.log(event);

    setCounter(counter + 1);
    // setCounter((c) => c + 1);
  };

  const handleSubtract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleSum}>Incrementar +</button>
      <button onClick={handleSubtract}>Disminuir -</button>
      <button aria-label='btn-reset' onClick={handleReset}>
        Resetear
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
