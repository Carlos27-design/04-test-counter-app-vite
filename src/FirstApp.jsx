import PropTypes from 'prop-types';

// const newMessage = {
//   message: 'Hola Mundo',
//   name: 'Carlos',
// };

//Se puede pero si la funcion no es asincrona
// const saludar = () => {
//   return 'Hola Mundo';
// };

// const mensaje = () => {
//   return 'Hola manito, Como estas Espero que estes bien no mÃ¡s ';
// };

// const numero = (a, b) => {
//   return a + b;
// };
export const FirstApp = ({
  title,
  subtitle = 'Todo el mundo explota po loco',
  name,
}) => {
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      {/*<code>{JSON.stringify(newMessage)}</code>*/}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

// Esto ayuda para ver como funciona nuestro component por sus Propertys
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

// Estos son Props por defecto
FirstApp.defaultProps = {
  //title: 'No hay titulos',
  subtitle: 'No hay subtitulo',
  name: 'Carlos Coronado',
};
