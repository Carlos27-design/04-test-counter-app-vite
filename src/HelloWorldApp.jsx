export const HelloWorldApp = () => {
  return <h1>HelloWorldApp</h1>;
};

//Formas con menos codigo
//export const HelloWorldApp = () => <h1>Arrow Function</h1>;
