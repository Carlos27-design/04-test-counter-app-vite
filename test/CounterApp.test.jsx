import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Pruebas a <CounterApp />', () => {
  //Teest: debe hacer match con el snapshop
  const value = 10;
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });

  //test: debe mostrar el valor inicial de 100
  test('Debe mostrar el valor inicial de 100', () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();
    // expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(
    //   '100'
    // );
  });

  test('debe de incrementar con el boton Incrementar +', () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText('Incrementar +'));
    expect(screen.getByText('11')).toBeTruthy();
  });

  test('debe de decrementar con el boton Disminuir -', () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText('Disminuir -'));
    expect(screen.getByText('9')).toBeTruthy();
  });

  test('debe de funcionar el boton de reset', () => {
    render(<CounterApp value={340} />);
    fireEvent.click(screen.getByText('Incrementar +'));
    fireEvent.click(screen.getByText('Incrementar +'));
    fireEvent.click(screen.getByText('Resetear'));

    // expect(screen.getByText(value)).toBeTruthy();

    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
    expect(screen.getByText(340)).toBeTruthy();
  });
});
