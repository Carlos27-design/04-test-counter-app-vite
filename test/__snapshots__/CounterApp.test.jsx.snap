// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
    Incrementar +
  </button>
  <button>
    Disminuir -
  </button>
  <button>
    Resetear
  </button>
</div>
`;
